test/e2e/client_test.go

```diff
@@ -106,3 +115,58 @@ func (t *testClient) cleanupProject(timeout time.Duration) error {
 	}
 	return nil
 }
+
+func (t *testClient) dumpInfo() error {
+	// gather events
+	eventList, err := t.kc.CoreV1().Events(t.namespace).List(metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+	eventBuf := bytes.NewBuffer(nil)
+	for _, event := range eventList.Items {
+		b, err := yaml.Marshal(event)
+		if err != nil {
+			return err
+		}
+		if _, err := eventBuf.Write(b); err != nil {
+			return err
+		}
+		if _, err := eventBuf.Write([]byte("\n")); err != nil {
+			return err
+		}
+	}
+
+	// gather pods
+	podList, err := t.kc.CoreV1().Pods(t.namespace).List(metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+	podBuf := bytes.NewBuffer(nil)
+	for _, pod := range podList.Items {
+		b, err := yaml.Marshal(pod)
+		if err != nil {
+			return err
+		}
+		if _, err := podBuf.Write(b); err != nil {
+			return err
+		}
+		if _, err := podBuf.Write([]byte("\n")); err != nil {
+			return err
+		}
+	}
+
+	if t.artifactDir != "" {
+		if err := ioutil.WriteFile(filepath.Join(t.artifactDir, fmt.Sprintf("events-%s.yaml", t.namespace)), eventBuf.Bytes(), 0777); err != nil {
+			logrus.Warn(err)
+			fmt.Println(eventBuf.String())
```

Comment:
why not `logrus.Info(eventBuf.String())`?

Reply:
I did a bare print here mainly because at this point we don't really care about the logging context and the yaml is going to be better formatted in the log.